### PR TITLE
Fix Profile: Parabolic Ch. Single Prec.

### PR DESCRIPTION
**Pull request by mrowan137**
This PR (a continuation of PR #1082) introduces a change to give better agreement between single  and double precision versions of the executable.  In particular, this PR changes the initialized density profile ('plasma initialization fix' in the plots below) in single precision; this fix seems to control the agreement at late times, independent of the fix in current deposition (this is  demonstrated in the plot below).

dp_sp_compare_Ex

The modifications here are not the minimal required changes; there is good agreement between SP and DP at late times by casting only `n0` as double, but for safety the other parameters are also cast as double in this PR.

Input file to reproduce runs from the videos is here: 
[inputs.txt](https://github.com/ECP-WarpX/WarpX/files/4935323/inputs.txt)

### CODE REVIEW

File: Source/Initialization/InjectorDensity.H [hunk 1]
```diff
@@ -65,14 +66,20 @@ struct InjectorDensityPredefined
...
+            // and double precision, but just in case, all are cast as double.
+            double z_start   = p[0];
+            double ramp_up   = p[1];
+            double plateau   = p[2];
+            double ramp_down = p[3];
+            double rc        = p[4];
+            double n0        = p[5];
+            double n;
+            double kp = PhysConst::q_e/PhysConst::c
                 *std::sqrt( n0/(PhysConst::m_e*PhysConst::ep0) );
```
Comment (ax3l) [inline Source/Initialization/InjectorDensity.H]: This line probably should use an amrex math function to be able to run on GPU (for the `sqrt`)

**Review by ax3l — COMMENTED** [review]
Awesome!
Can you please add a few python lines to the test where you saw this (`analysis.py`)? With that you can ensure the density is initialized correctly at t=0 (or t=1 or so).

File: Source/Initialization/InjectorDensity.H (hunk 1 above)
Comment (mrowan137) [reply Source/Initialization/InjectorDensity.H]: Fixed.

**Comment by mrowan137** [comment]
> Awesome!
> Can you please add a few python lines to the test where you saw this (`analysis.py`)? With that you can ensure the density is initialized correctly at t=0 (or t=1 or so).

Added a checksum regression test, which compares the initialized profile in SP to the DP result.

**Review by ax3l — COMMENTED** [review]
Thank you!

**Review by ax3l — APPROVED** [review]

**Pull request merged**